gprofiler/metadata/application_identifiers_java.py

```diff
@@ -11,16 +11,19 @@
 from gprofiler.exceptions import CalledProcessError
 from gprofiler.log import get_logger_adapter
 from gprofiler.metadata.base_application_identifier import _ApplicationIdentifier
-from gprofiler.profilers.java import jattach_path
-from gprofiler.utils import run_process
+from gprofiler.profilers.java import JattachJcmdRunner
 
 _logger = get_logger_adapter(__name__)
 
 
 class _JavaJarApplicationIdentifier(_ApplicationIdentifier):
+    def __init__(self, jattach_jcmd_runner: JattachJcmdRunner):
```

Comment:
Missing call to `super()` c'tor

Reply:
but `_ApplicationIdentifier` is an abstract class that doesn't define an `__init__`, why should `_JavaJarApplicationIdentifier` call `super.__init__()`?

Reply:
It's not abstract, it's an empty class with no c'tor definition, hence the default one.

If you don't leave a call to the c'tor here, then at some point in the future, a c'tor might be added to the parent class, and this new class will misbehave because you don't call the c'tor. It's the best practice to call it anyway.